ruby/src/IceRuby/Communicator.cpp

```diff
@@ -25,12 +25,12 @@ using CommunicatorMap = map<Ice::CommunicatorPtr, VALUE>;
 static CommunicatorMap _communicatorMap;
 
 extern "C" void
-IceRuby_Communicator_mark(Ice::CommunicatorPtr* p)
+IceRuby_Communicator_mark(void* p)
```

Comment:
In the previous code, the `Data_Wrap_Struct` macro was casting these functions. I updated these functions to use the correct signature and do a static cast to the expected type in the implementation.